bokeh/embed/bundle.py

```diff
@@ -414,7 +414,18 @@ def _use_mathjax(objs: Sequence[Model | Document]) -> bool:
         bool
     '''
     from ..models.text import MathText
-    return _any(objs, lambda obj: isinstance(obj, MathText))
+
+    def model_has_mathstring(model: Model) -> bool:
+        ''' Whether any of the properties of the given model accepts a `MathText` object and has given a mathstring.
+        This is needed because we convert mathstrings to `MathText` on BokehJS views.
+        '''
+        properties = model.properties(_with_props = True)
+
+        for key, value in model._property_values.items():
+            if properties[key].is_valid(MathText()) and MathText.is_math_text_string(value):
```

Comment:
This is actually fairly expensive operation, and one reason I can't say I love this approach. It still makes much more sense to me for a dedicated property type to encapsulate the detection and handling of mathtext, and I still have not heard any rationale for why we veered from that course.

Reply:
Edit: to clarify, I mean *on the python side. Nothind we do no the python side precludes adding separate conveniences on the JS side, but one need not dictate the other.

Reply:
The problem with the previous approach with transforms is that some properties inside other properties (#11545 ) don't transform like expected. And the problem with accepts is that it expects every property to fail validation before it converts the value, so we would not be able to keep accepting strings and convert strings to MathText

Reply:
One possible solution would be removing plain strings support from models' properties that implement MathText

Reply:
> The problem with the previous approach with transforms is that some properties inside other properties (#11545 ) don't transform like expected. 

We had a solution proposed for that: a semantically meaningful type for the one case we know about now, which could be generalized in the future, if absolutely necessary. What was the issue with that?

> And the problem with accepts is that it expects every property to fail validation before it converts the value, so we would not be able to keep accepting strings and convert strings to MathText

There is also an issue that describes a solution to this (i.e. add a `converts` to go along with `accepts` that happens before validation. That is a fairly trivial addition unless I am mistaken. 

> One possible solution would be removing plain strings support from models' properties that implement MathText. 

Definitely a non-starter

Reply:
> We had a solution proposed for that: a semantically meaningful type for the one case we know about now, which could be generalized in the future, if absolutely necessary. What was the issue with that?

I don't remember exactly what was blocking me on the `DictWithMathstringTransform` but I would prefer the `converts` method anyway. Let me see how hard it is

> Definitely a non-starter

Alright, but IMO it COULD be useful to use models instead of plain strings in some places, no time for this discussion now but I will keep notes if I find those places.